autopts/utils.py

```diff
@@ -477,6 +479,61 @@ def log_memory_usage():
     logging.debug(f"Memory usage: {mem_usage:.2f} MB")
 
 
+def extract_wid_testcases_to_csv():
+    log_dir = "./logs"
+    output_csv_path = "./wid_usage_report.csv"
+    profile_wid_map = defaultdict(lambda: defaultdict(set))
+
+    for root, _, files in os.walk(log_dir):
+        for filename in files:
+            if filename.endswith(".log"):
+                filepath = os.path.join(root, filename)
+                with open(filepath, encoding='utf-8', errors='ignore') as f:
+                    lines = f.readlines()
+
+                in_block = False
```

Comment:
Can you elaborate on what `in_block` means?

Reply:
It indicates whether we’re currently parsing inside a "BEGIN OnImplicitSend:" ... "END OnImplicitSend" log block. I also renamed it to in_implicit_send_block for clarity.